reconcile/templating/lib/model.py

```diff
@@ -1,34 +1,23 @@
-from typing import Any
-
 from deepdiff import DeepHash
 from pydantic import BaseModel
 
-from reconcile.gql_definitions.templating.template_collection import (
-    TemplateV1,
-)
+
+class TemplateOutput(BaseModel):
+    is_new: bool = False
+    path: str
+    content: str
+    auto_approved: bool = False
 
 
-class TemplateInput(BaseModel):
+class TemplateResult(BaseModel):
     collection: str
-    templates: list[TemplateV1] = []
-    variables: list[dict[str, Any]] = []
-    collection_hash: str = ""
+    result_hash: str = ""
     enable_auto_approval: bool = False
     labels: list[str] = []
+    outputs: list[TemplateOutput] = []
 
     def calc_template_hash(self) -> str:
-        if not self.collection_hash:
-            hashable = {
-                "templates": sorted(self.templates, key=lambda x: x.name),
-                "variables": self.variables,
-            }
-            self.collection_hash = DeepHash(hashable)[hashable]
-        return self.collection_hash
-
-
-class TemplateOutput(BaseModel):
-    input: TemplateInput
-    is_new: bool = False
-    path: str
-    content: str
-    auto_approved: bool = False
+        if not self.result_hash:
+            hashable = {"outputs": sorted(self.outputs, key=lambda x: x.path)}
+            self.result_hash = DeepHash(hashable)[hashable]
+        return self.result_hash
```

Comment:
This method is to calc hash only once, right? Can use [cached_property](https://docs.python.org/3/library/functools.html#functools.cached_property) to simplify it, example to use with `BaseModel` at https://github.com/app-sre/qontract-reconcile/blob/1b80a586850a1145aff072f7837b71bb716d223d/reconcile/utils/early_exit_cache.py#L76.

To make hash safe and stable, should also mark `TemplateOutput` as frozen. If `path` is unique, can make `hashable` a dict `{o.path: o for o in self.outputs}`

Reply:
I found out calc hash would only run once when submitting MR, so there is no need to add cache.